src/app/guards/load-streams.guard.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree } from '@angular/router';
+import {combineLatest, Observable, of} from 'rxjs';
+import {Store} from '@ngrx/store';
+import {loadStreams} from '../store/stream/stream.actions';
+import {filter, take, tap, switchMap} from 'rxjs/operators';
+import {getIsStreamLoaded, getIsStreamLoading} from '../store/stream/stream.selectors';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoadStreamsGuard implements CanActivate {
+  constructor(private store: Store) {}
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+
+    const isStreamsLoading$ = this.store.select(getIsStreamLoading);
```

Comment:
A good choice, but since you are only using it once, you can simply move that logic to combineLatest :)
Up to your choice to leave it or not :)